plugins/heroku/src/setConfigVars.ts

```diff
@@ -25,15 +27,46 @@ async function setConfigVars(
     await heroku.patch(`/apps/${appIdName}/config-vars`, { body: configVars })
 
     logger.verbose('the following values have been set:', Object.keys(configVars).join(', '))
-
+    
     logger.info(`${appIdName} config vars have been updated successfully.`)
+
+  } catch (err) {
+    const error = new ToolKitError(`Error updating config vars for ${appIdName} app`)
+    if (err instanceof Error) {
+      error.details = err.message
+    }
+    throw error
+  }
+}
+
+async function setStageConfigVars(
+  logger: Logger,
+  stage: Stage,
```

Comment:
**praise**: keeping it flexible like this is really nice (as well as using the union type)!